wetlab/views.py

```diff
@@ -2441,230 +2410,119 @@ def record_samples(request):
                 "wetlab/record_sample.html",
                 {
                     "fields_info": fields_info,
-                    "error_message": "Excell file is empty",
+                    "error_message": core.core_config.ERROR_EMPTY,
                 },
             )
         else:
             # Check if all samples have same project and if project is in the DB
-            if core.utils.load_batch.project_validation(sample_batch_df, __package__):
+            validation_project, project_name = core.utils.load_batch.project_validation(
+                sample_batch_df, __package__
+            )
+            if validation_project["result"] == "NOK":
                 return render(
                     request,
                     "wetlab/record_sample.html",
                     {
                         "fields_info": fields_info,
-                        "error_message": core.utils.load_batch.project_validation(
-                            sample_batch_df, __package__
-                        ),
+                        "error_message": validation_result["error_message"],
                     },
                 )
-
-            sample_batch_df = core.utils.load_batch.heading_refactor(sample_batch_df)
```

Comment:
do not remove this function. Also one thing I didn't notice when Sarai did this, this function must use the fields_info dictionary  (which is global for "record_samples" view) instead of calling again to the meta fields in the Sample table.

Reply:
I did not removed. I just move down after doing the validation